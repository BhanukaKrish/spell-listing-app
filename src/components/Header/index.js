import React from "react";
import style from "./style.module.css";
import Logo from "../../assets/images/DnD-Logo.png";
import { Link } from "react-router-dom";

function Header() {
  return (
    <div className={style.header_bar}>
      <div className="container">
        <div className={style.header}>
          <div className={style.header__logo}>
            <img src={Logo} alt="D&D Logo" />
          </div>
          <div className={style.header__right}>
            <ul className={style.header__menu}>
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                <Link to="/favourites">favourites</Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Header;
