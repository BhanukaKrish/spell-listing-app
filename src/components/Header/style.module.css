.header_bar {
  top: 0;
  position: sticky;
  z-index: 100;
  background-color: #000;
  box-shadow: 0px 2px 4px -1px rgb(0 0 0 / 20%),
    0px 4px 5px 0px rgb(0 0 0 / 14%), 0px 1px 10px 0px rgb(0 0 0 / 12%);
}

.header {
  display: flex;
  align-items: center;
  width: 100%;
  height: 70px;
  justify-content: space-between;
}

.header__logo {
  width: 120px;
}

.header__menu {
  display: flex;
  align-items: center;
}

.header__menu > li {
  margin-left: 20px;
}

.header__menu > li > a {
  color: var(--color-white);
  text-decoration: none;
  font-size: var(--font-size-base);
  font-weight: 300;
  transition: var(--transition);
}

.header__menu > li > a:hover {
  color: var(--color-primary);
}
