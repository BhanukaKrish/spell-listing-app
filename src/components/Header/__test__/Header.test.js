import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "../Header";
import { MemoryRouter } from "react-router-dom";

describe("Test the Header Component", () => {
  test("header should have a logo", () => {
    render(<Header />, { wrapper: MemoryRouter });
    const logo = screen.getByRole("img");
    expect(logo).toBeInTheDocument();
  });

  test("header should have a Home menu", () => {
    render(<Header />, { wrapper: MemoryRouter });
    const menu = screen.getByText("Home");
    expect(menu).toBeInTheDocument();
  });

  test("header should have a Favourites menu", () => {
    render(<Header />, { wrapper: MemoryRouter });
    const menu = screen.getByText("Favourites");
    expect(menu).toBeInTheDocument();
  });

  test("Snapshot of Header", () => {
    const { container } = render(<Header />, { wrapper: MemoryRouter });
    expect(container).toMatchSnapshot();
  });
});
