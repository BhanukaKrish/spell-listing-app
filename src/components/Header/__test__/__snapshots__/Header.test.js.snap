// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Header Component Snapshot of Header 1`] = `
<div>
  <div
    class="header_bar"
  >
    <div
      class="container"
    >
      <div
        class="header"
      >
        <div
          class="header__logo"
        >
          <img
            alt="D&D Logo"
            src="DnD-Logo.png"
          />
        </div>
        <div
          class="header__right"
        >
          <ul
            class="header__menu"
          >
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/favourites"
              >
                Favourites
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
